pkg/domain/infra/abi/volumes.go

```diff
@@ -154,6 +155,24 @@ func (ic *ContainerEngine) VolumeList(ctx context.Context, opts entities.VolumeL
 	return reports, nil
 }
 
+func (ic *ContainerEngine) generateVolumeListFilters(opts entities.VolumeListOptions) ([]libpod.VolumeFilter, error) {
+	var (
+		volumeFilters []libpod.VolumeFilter
+		err           error
+	)
+	// special internal request handler to get all until=time
+	// volumes using prune filters instead of list filters
+	if opts.UntilCmdRequest {
```

Comment:
Added this one to not expose until in list filters portfolio and keep the current architecture (internal request possibility) in cmd volumes prune. I personally think there is no better way to do this currently, but I am open to suggestions.

Reply:
Is there any reason why until should not be allowed for list?

Reply:
@Luap99 yes. It is an extension as in docker it does not exist. If you accept this as compat extension or compatibility breakage it's fine. If not, we should stick with the proposed solution. Let me know what you think.

Reply:
I don't see any problems adding until to list. Adding more filters is not a compatibility breakage for me. I mean who is going to call list with the until filter and expect it to fail.
But let's wait what other maintainers think, @rhatdan @mheon @vrothberg?

Reply:
Not a problem @Luap99. Let's hear other comments and if so, i can add one more filter. I just followed docker example here.

Reply:
I am cool if the compat layer can do more than Docker.

Reply:
Yeah, our big concern is that we don't do less than they do, and what we both do should match Docker's behavior closely. If our compat code supports extra options, it's not a big issue.

Reply:
@mheon @Luap99 @vrothberg https://github.com/containers/podman/pull/11015 was created as prerequisite to this PR

Reply:
Done